src/main/kotlin/org/jetbrains/research/libsl/visitors/AutomatonResolver.kt

```diff
@@ -22,17 +23,41 @@ class AutomatonResolver(
         val name = ctx.name.asPeriodSeparatedString()
         val typeName = ctx.type.asPeriodSeparatedString()
         val typeReference = TypeReferenceBuilder.build(typeName, context = context)
+        val annotationReferences = makeAutomatonAnnotationList(ctx.automatonAnnotations())
 
         buildingAutomaton = Automaton(
             name,
             typeReference,
+            annotationReferences,
             context = automatonContext
         )
 
         super.visitAutomatonDecl(ctx)
         context.parentContext!!.storeAutomata(buildingAutomaton)
     }
 
+    private fun makeAutomatonAnnotationList(ctx: List<LibSLParser.AutomatonAnnotationsContext>?): MutableList<AnnotationReference>? {
```

Comment:
Please, use more conventional name patterns like `getAutomatonAnnotationList`, `createAutomatonAnnotationList` or `buildAutomatonAnnotationList`

(I'd use the first one here)

Reply:
Also, may be you should make every property contains annotations in `node`s non-nullable. To show that there are no annotations you could use an empty collection

Reply:
Nullable collections is a code smell